src/NServiceBus.Core.Tests/Routing/MessageDrivenSubscriptions/AssemblyPublisherSourceTests.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace NServiceBus.Core.Tests.Routing
+{
+    using System.Linq;
+    using System.Reflection;
+    using NServiceBus.Routing.MessageDrivenSubscriptions;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class AssemblyPublisherSourceTests
+    {
+        [Test]
+        public void It_returns_only_event_types()
+        {
+            var source = new AssemblyPublisherSource(Assembly.GetExecutingAssembly(), PublisherAddress.CreateFromEndpointName("Destination"));
+            var routes = source.Generate(new Conventions()).ToArray();
```

Comment:
No opt in. But this only applies to the code-first APIs. The MEMs work like they did. Should we allow non-events to be configured for publishing with the code-firs API?